src/main/kotlin/org/rust/lang/core/types/infer/Expressions.kt

```diff
@@ -127,10 +127,24 @@ fun inferExpressionType(expr: RsExpr): RustType {
                 RustUnknownType
         }
 
+        is RsArrayExpr -> inferArrayType(expr)
+
         else -> RustUnknownType
     }
 }
 
+private fun inferArrayType(expr: RsArrayExpr): RustType {
+    val (elementType, size) = if (expr.semicolon != null) {
+        if (expr.exprList.size != 2) return RustUnknownType
+        val (elementExpr, sizeExpr) = expr.exprList
+        val size = calculateArraySize(sizeExpr) ?: return RustUnknownType
```

Comment:
At this moment I can't say what it will affect and that will be better because I'm still not very familiar with the internal type system.
Let's use `RustSliceType` in such ambiguous cases for now and if you consider that `RsArrayType` and `RustSliceType` must be represented with one class we will merge them in other PR